src/main/java/io/github/ryotan/code/CodeEnum.java

```diff
@@ -0,0 +1,274 @@
+     *
+     * @return {@link Enum#ordinal()}
+     */
+    int ordinal();
+
+    /**
+     * コードが{@code codes}で与えられたコードに含まれているかどうかを返します。
+     *
+     * @param codes 判定対象のコードの集合
+     * @return コードが {@code codes} に含まれる場合 {@code true}
+     */
+    @SuppressWarnings("unchecked")
+    default boolean in(C... codes) {
+        return Stream.of(codes).anyMatch(this::equals);
+    }
+
+    /**
+     * コードが{@code codes}で与えられたコードに含まれないかどうかを返します。
+     *
+     * @param codes 判定対象のコードの集合
+     * @return コードが {@code codes} に含まれない場合 {@code true}
+     */
+    @SuppressWarnings("unchecked")
+    default boolean not(C... codes) {
+        return !this.in(codes);
+    }
+
+    /**
+     * {@link CodeEnum}のフィルタリングをする{@link Predicate}を楽に生成できるようにするユーティリティクラスです。
+     */
+    final class Filters {
+
+        /**
+         * 任意の{@link CodeEnum}に対して{@code true}を返します。
+         */
+        public static final Predicate<? super CodeEnum<?>> ANY = c -> true;
+
+        /**
+         * hidden constructor for utility class.
+         */
+        private Filters() {
+        }
+
+        /**
+         * {@code includes}に与えられたコードに含まれるかどうかを判定する{@link Predicate}を返します。
+         *
+         * @param includes 判定対象のコードの集合
+         * @param <T>      判定対象のコードの型
+         * @return {@code includes}に含まれる{@code T}のEnumが渡されたときに{@code true}を返す{@link Predicate}
+         */
+        @SafeVarargs
+        public static <T extends CodeEnum<T>> Predicate<T> include(T... includes) {
+            return c -> Stream.of(includes).anyMatch(c::equals);
+        }
+
+        /**
+         * {@code excludes}に与えられたコードに含まれないかどうかを判定する{@link Predicate}を返します。
+         *
+         * @param excludes 判定対象のコードの集合
+         * @param <T>      判定対象のコードの型
+         * @return {@code excludes}に含まれない{@code T}のEnumが渡されたときに{@code true}を返す{@link Predicate}
+         */
+        @SafeVarargs
+        public static <T extends CodeEnum<T>> Predicate<T> exclude(T... excludes) {
+            return include(excludes).negate();
+        }
+    }
+
+    /**
+     * コードをフィルタリングするための関数を表すマーカーインターフェースです。
+     * <p>
+     * 次のように定義することが出来ます。
+     * </p>
+     * <pre><code class="java">
+     * </code></pre>
+     * <p>
+     * 次のように利用することで、フィルタリングされたコードの集合を取得できます。
+     * </p>
+     * <pre><code class="java">
+     * </code></pre>
+     * <p>
+     * また、フィルタリングされたコードの集合に含まれるかどうかを判定することも出来ます。
+     * </p>
+     * <pre><code class="java">
+     * </code></pre>
+     */
+    @Target({METHOD, FIELD})
+    @Retention(RUNTIME)
+    @Documented
+    @interface Filter {
+    }
+
+    /**
+     * コードの論理名の短縮名を表すために利用するアノテーションです。
+     * （とあるFWのためだけに用意されています。別名が必要な場合は、{@link AliasLabel} を利用したほうが統一感があるので良いと思います。）
+     * <p>
+     * 次のように、Enum定数に対してアノテーションを付けることで、短縮論理名を設定することが出来ます。
+     * </p>
+     * <pre><code class="java">
+     * &#64;ShortLabel("男")
+     * MALE("1", "男性"),
+     * </code></pre>
+     * <p>
+     * なお、アノテーションを付けるフィールドは{@link String}型でなくてはいけません。
+     * </p>
+     */
+    @Target({FIELD})
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Annotation style must be 'COMPACT_NO_ARRAY'.](https://www.codacy.com/app/ryotan/code-enum-support/file/2914735243/issues/source?bid=3375378&fileBranchId=3375797#l226)